### PR TITLE
feat: Add Russian translation to firmware 🇷🇺 

### PR DESCRIPTION
**Pull request by kobakirill**
Added Russian translation for devices. Added fonts for both black and white and color screens.
Work has been carried out to reduce the words. A lot of 
routine work. Tested on tx12/tx16S
photo_2023-09-19_23-41-19
photo_2023-09-20_00-02-41
photo_2023-10-07_01-34-16
photo_2023-10-07_01-34-25
photo_2023-10-07_01-34-32
photo_2023-10-07_01-34-37
photo_2023-10-07_01-34-42

ZORRO old font
photo_2023-10-05_14-52-47 (2)

### CODE REVIEW

File: radio/src/thirdparty/FatFs/ffconf.h [hunk 1]
```diff
@@ -78,8 +78,10 @@
 /*---------------------------------------------------------------------------/
 / Locale and Namespace Configurations
 /---------------------------------------------------------------------------*/
-
-#define FF_CODE_PAGE	437
+#if defined(TRANSLATIONS_RU)
+#else
+#define _CODE_PAGE	437
```
Comment (rotorman) [inline radio/src/thirdparty/FatFs/ffconf.h:83]: You removed define for FF_CODE_PAGE ?

Comment (kobakirill) [reply radio/src/thirdparty/FatFs/ffconf.h:83]: Sorry this is my mistake from the experiments

File: radio/src/thirdparty/FatFs/ffconf.h [hunk 2]
```diff
@@ -79,7 +79,7 @@
 / Locale and Namespace Configurations
 /---------------------------------------------------------------------------*/
 
-#define FF_CODE_PAGE	437
+#define FF_CODE_PAGE 437
```
Comment (rotorman) [inline radio/src/thirdparty/FatFs/ffconf.h:82]: Unnecessary white space change here

Comment (kobakirill) [reply radio/src/thirdparty/FatFs/ffconf.h]: /---------------------------------------------------------------------------*/


#define FF_CODE_PAGE    437
/* This option specifies the OEM code page to be used on the target system.

**Review by kobakirill — COMMENTED** [review]
reduction

File: tools/charset.py [hunk 3]
```diff
@@ -15,7 +15,7 @@ def is_special_char(c):
 
 def get_special_chars():
   result = {}
-  for lang in["cn", "cz", "da", "de", "en", "es", "fi", "fr", "he", "it", "jp", "nl", "pl", "pt", "se", "tw"]:
+  for lang in["cn", "cz", "da", "de", "en", "es", "fi", "fr", "he", "it", "jp", "nl", "pl", "pt","ru","se", "tw"]:
```
Comment (rotorman) [inline tools/charset.py:18]: Minor cosmetic issue - spaces missing before and after "ru" here.

Comment (kobakirill) [reply tools/charset.py]: cosmetic fix

**Comment by kobakirill** [comment]
ZORRO
photo_2023-10-05_14-52-47 (2)
phot
![photo_2023-10-05_15-22-20
o_2023-10-05_14-52-47](https://github.com/EdgeTX/edgetx/assets/39222879/1e4fb1e5-cdea-4f0c-9875-9570dfb64562)

**Comment by Roman-Golod** [comment]
I think it 'd be useful for a lot of russuan users, especially novices. Great work, thanks.

**Comment by Coffemansky** [comment]
Beautiful. The Russian language has been needed for a long time

**Review by rotorman — APPROVED** [review]

**Review by pfeerick — APPROVED** [review]
Thank you for all your hard work on this, I am sure the Russian native language speakers will greatly appreciate it! :)

**Comment by pfeerick** [comment]
Are you happy for this to merge in the next 2-3 days (probably the day after #4194)? 

You can also add
```
#define TR_TRAINER_STILL_CONNECTED     "Trainer still connected"
```
above
```
#define TR_USB_STILL_CONNECTED         "USB still connected"
```

in advance of #4212 merging if you want. 

Is there any particular reason #4195 isn't part of this PR, since this also touches on Companion translations (which is perfectly fine). 

Is it ok to add you to the list to get pinged for translations (for `RU`)?

**Comment by kobakirill** [comment]
> Are you happy for this to merge in the next 2-3 days (probably the day after #4194)?
> 
> You can also add
> 
> ```
> #define TR_TRAINER_STILL_CONNECTED     "Trainer still connected"
> ```
> 
> above
> 
> ```
> #define TR_USB_STILL_CONNECTED         "USB still connected"
> ```
> 
> in advance of #4212 merging if you want.
> 
> Is there any particular reason #4195 isn't part of this PR, since this also touches on Companion translations (which is perfectly fine).
> 
> Is it ok to add you to the list to get pinged for translations (for `RU`)?

#define TR_MODEL_STILL_POWERED         "Модель еще работает"
#define TR_TRAINER_STILL_CONNECTED     "Тренер еще подкл"

**Comment by kobakirill** [comment]
> Are you happy for this to merge in the next 2-3 days (probably the day after #4194)? 
> 
> You can also add
> ```
> #define TR_TRAINER_STILL_CONNECTED     "Trainer still connected"
> ```
> above
> ```
> #define TR_USB_STILL_CONNECTED         "USB still connected"
> ```
> 
> in advance of #4212 merging if you want. 
> 
> Is there any particular reason #4195 isn't part of this PR, since this also touches on Companion translations (which is perfectly fine). 
> 
> Is it ok to add you to the list to get pinged for translations (for `RU`)? 
> 
> 

 
> Is it ok to add you to the list to get pinged for translations (for `RU`)? 
yes

**Comment by richardclli** [comment]
@kobakirill You should not merge main to you branch for PR, it is always better to use the "rebase" for this. e.g. rebase a branch to main = using main as base and try to commit all changes one by one, and you may need to resolve any conflicts during rebase.  And after doing the rebase, you need to check if everything alright and do a "force" push to your branch.

**Comment by kobakirill** [comment]
> @kobakirill You should not merge main to you branch for PR, it is always better to use the "rebase" for this. e.g. rebase a branch to main = using main as base and try to commit all changes one by one, and you may need to resolve any conflicts during rebase. And after doing the rebase, you need to check if everything alright and do a "force" push to your branch.

OK

**Pull request merged**